biogeophys/FatesHydroWTFMod.F90

```diff
@@ -210,6 +210,7 @@ function th_from_psi_base(this,psi) result(th)
     class(wrf_type)     :: this
     real(r8),intent(in) :: psi
     real(r8)            :: th
+    th = 0._r8
```

Comment:
This is just to avoid IBM compiler warning with `th` as uninitialized function result:
```
"/autofs/nccs-svm1_home1/azamat/repos/E3SM-tests/components/clm/src/external_models/fates/biogeophys/FatesHydroWTFMod.F90", 1513-083 (E) Internal or module function ftc was not set within the function.
"/autofs/nccs-svm1_home1/azamat/repos/E3SM-tests/components/clm/src/external_models/fates/biogeophys/FatesHydroWTFMod.F90", 1513-083 (E) Internal or module function dftcdpsi was not set within the function.
"/autofs/nccs-svm1_home1/azamat/repos/E3SM-tests/components/clm/src/external_models/fates/biogeophys/FatesHydroWTFMod.F90", 1513-083 (E) Internal or module function dpsidth was not set within the function.
"/autofs/nccs-svm1_home1/azamat/repos/E3SM-tests/components/clm/src/external_models/fates/biogeophys/FatesHydroWTFMod.F90", 1513-083 (E) Internal or module function th was not set within the function.
"/autofs/nccs-svm1_home1/azamat/repos/E3SM-tests/components/clm/src/external_models/fates/biogeophys/FatesHydroWTFMod.F90", 1513-083 (E) Internal or module function psi was not set within the function.
```